### PR TITLE
Create a BagIt bag at the end of preprocessing

### PR DESCRIPTION
**Pull request by djjuhasz**

### CODE REVIEW

**Comment by djjuhasz** [comment]
@jraddaoui I'm not sure if we want to add the "CreateBagActivity" here, or just add it to preprocessing-sfa and preprocessing-moma?

**Comment by jraddaoui** [comment]
Thanks @djjuhasz! I kind of like having it here as an example, even if not all child workflows will need it. It also gives us the chance to improve the readme and the requirements to make it work with Enduro, I'll follow-up with that once this is merged.

**Review by jraddaoui — COMMENTED** [review]
Nice!

Still having my conflicts about how the config/params are handled in the activity, as noted in https://github.com/artefactual-sdps/temporal-activities/pull/16, but this looks good to me.

File: internal/config/config.go [hunk 1]
```diff
@@ -30,7 +35,16 @@ type WorkerConfig struct {
 	MaxConcurrentSessions int
 }
 
-func (c Configuration) Validate() error { return nil }
+func (c Configuration) Validate() error {
+	var errs error
+
+	validate := validator.New(validator.WithRequiredStructEnabled())
```
Comment (jraddaoui) [inline internal/config/config.go:41]: Can we create the validator inside the `Validate` function to avoid the dependency here for now?

Comment (djjuhasz) [reply internal/config/config.go:41]: @jraddaoui oops, I didn't see your "thumbs up" until now. :facepalm: 

If we merge #5 first, then I can remove the `bagit.Config.Validate()` method altogether, and just run the validator here against the `bagit.Config` validation struct tag.

Comment (jraddaoui) [reply internal/config/config.go:41]: Yeah, thumbs up is probably not the best way to provide feedback, sorry ;) Check my comments in the other PR.

**Comment by djjuhasz** [comment]
I tested the preprocessing workflow, and bagging works... kind of.  :(

The first time a transfer is submitted, a bag is created in place.  If the same directory is submitted again, then bagging fails with an error:

> CreateBagActivity: create bag: mkdir /home/preprocessing/shared/small/data: file exists

This error will occur any time a transfer with the same "relative path" (e.g. "small") is submitted, because we are modifying the transfer in place. :(

I think we are going to need to create a unique working directory for each workflow instance to prevent errors on duplicate submissions and to prevent failed preprocessing workflows from leaving the source transfer in a partially transformed state.

**Comment by djjuhasz** [comment]
> The first time a transfer is submitted, a bag is created in place. If the same directory is submitted again, then bagging fails with an error:
> 
> > CreateBagActivity: create bag: mkdir /home/preprocessing/shared/small/data: file exists

The above error doesn't occur when preprocessing is run as a child workflow of the Enduro processing workflow - Enduro creates a unique local transfer directory for each workflow run.

**Review by jraddaoui — COMMENTED** [review]
Looking great!

**Review by jraddaoui — APPROVED** [review]
Neat!

**Pull request merged**